packages/babel-generator/src/printer.ts

```diff
@@ -427,6 +428,7 @@ class Printer {
 
     this._endsWithWord = false;
     this._endsWithInteger = false;
+    this._endsWithDiv = str === "/";
```

Comment:
Can we do this to avoid comparing everything?
`this._endsWithDiv = false`
```ts
export function AssignmentExpression(
  this: Printer,
  node: t.AssignmentExpression | t.LogicalExpression | t.BinaryExpression,
) {
  this.print(node.left, node);

  this.space();
  if (node.operator === "in" || node.operator === "instanceof") {
    this.word(node.operator);
  } else {
    if (node.operator === "/") {
      this.tokenChar(47);
    } else {
      this.token(node.operator);
    }
  }
  this.space();

  this.print(node.right, node);
}
```